### PR TITLE
Enable counter based autoscaler to scale from 0 replicas

### LINKED ISSUE #4047: Fleet autoscaler with "Counter" policy throws an error if configured with a fleet with no replicas
This is basically #3943, but for Counters, so check there for details 😀 

PR https://github.com/googleforgames/agones/pull/4016 solved this for Lists, but not for Counters, so it should hopefully be pretty straight forward to replicate the logic for Counters.

### PR DESCRIPTION
**Pull request by geopaulm**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
 /kind bug
> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:
Enables counter based autoscaler to scale up correctly when there are 0 replicas.

**Which issue(s) this PR fixes**:

Closes #4047

**Special notes for your reviewer**:

### CODE REVIEW

File: test/e2e/fleetautoscaler_test.go [hunk 1]
```diff
@@ -944,6 +944,87 @@ func TestCounterAutoscaler(t *testing.T) {
...
+	}
+
+	flt, err := client.Fleets(framework.Namespace).Create(ctx, flt.DeepCopy(), metav1.CreateOptions{})
+	require.NoError(t, err)
+	defer client.Fleets(framework.Namespace).Delete(ctx, flt.ObjectMeta.Name, metav1.DeleteOptions{}) // nolint:errcheck
+	framework.AssertFleetCondition(t, flt, e2e.FleetReadyCount(flt.Spec.Replicas))
+
+	fleetautoscalers := framework.AgonesClient.AutoscalingV1().FleetAutoscalers(framework.Namespace)
+
+	listFas := func(f func(fap *autoscalingv1.FleetAutoscalerPolicy)) *autoscalingv1.FleetAutoscaler {
```
Comment (peterzhongyi) [inline test/e2e/fleetautoscaler_test.go:971]: should it be counterFas?

**Review by peterzhongyi — APPROVED** [review]
LGTM

**Comment by peterzhongyi** [comment]
Ran the tests locally, and verified the behaviors by manually incrementing and decrementing the counter on the simple game server to see it scaling up and down.

File: test/e2e/fleetautoscaler_test.go [hunk 2]
```diff
@@ -944,6 +944,88 @@ func TestCounterAutoscaler(t *testing.T) {
...
+	flt, err := client.Fleets(framework.Namespace).Create(ctx, flt.DeepCopy(), metav1.CreateOptions{})
+	require.NoError(t, err)
+	defer client.Fleets(framework.Namespace).Delete(ctx, flt.ObjectMeta.Name, metav1.DeleteOptions{}) // nolint:errcheck
+	framework.AssertFleetCondition(t, flt, e2e.FleetReadyCount(flt.Spec.Replicas))
+
+	fleetautoscalers := framework.AgonesClient.AutoscalingV1().FleetAutoscalers(framework.Namespace)
+
+	counterFas := func(f func(fap *autoscalingv1.FleetAutoscalerPolicy)) *autoscalingv1.FleetAutoscaler {
+		fas := autoscalingv1.FleetAutoscaler{
+			ObjectMeta: metav1.ObjectMeta{Name: flt.ObjectMeta.Name + "-list-autoscaler", Namespace: framework.Namespace},
```
Comment (igooch) [inline test/e2e/fleetautoscaler_test.go:974]: Nit: 
```suggestion
			ObjectMeta: metav1.ObjectMeta{Name: flt.ObjectMeta.Name + "-counter-autoscaler", Namespace: framework.Namespace},
```

Comment (geopaulm) [reply test/e2e/fleetautoscaler_test.go]: oh no!. Fixed!

**Review by igooch — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**